Index: amqp-cpp/CMakeLists.txt
===================================================================
--- amqp-cpp.orig/CMakeLists.txt
+++ amqp-cpp/CMakeLists.txt
@@ -19,9 +19,11 @@ set (VERSION_MINOR 3)
 set (VERSION_PATCH 27)
 set (SO_VERSION ${VERSION_MAJOR}.${VERSION_MINOR})
 
+EXECUTE_PROCESS(COMMAND dpkg-architecture -qDEB_HOST_MULTIARCH OUTPUT_VARIABLE CMAKE_ARCH_TRIPLET OUTPUT_STRIP_TRAILING_WHITESPACE)
+
 # build options
-option(AMQP-CPP_BUILD_SHARED "Build shared library. If off, build will be static." OFF)
-option(AMQP-CPP_LINUX_TCP "Build linux sockets implementation." OFF)
+option(AMQP-CPP_BUILD_SHARED "Build shared library. If off, build will be static." ON)
+option(AMQP-CPP_LINUX_TCP "Build linux sockets implementation." ON)
 option(AMQP-CPP_BUILD_EXAMPLES "Build amqpcpp examples" OFF)
 
 # pass version number to source files as macro
@@ -115,7 +117,8 @@ install(DIRECTORY include/amqpcpp/ DESTI
         FILES_MATCHING PATTERN "*.h")
 install(FILES include/amqpcpp.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
-install(EXPORT ${PROJECT_NAME}Config DESTINATION cmake)
+install(EXPORT ${PROJECT_NAME}Config DESTINATION
+  ${CMAKE_INSTALL_LIBDIR}/cmake/amqpcpp)
 export(TARGETS ${PROJECT_NAME} FILE ${PROJECT_NAME}Config.cmake)
 
 set(DEST_DIR "${CMAKE_INSTALL_PREFIX}")
